addons/website_sale/static/src/js/website_sale.js

```diff
@@ -673,15 +705,19 @@ publicWidget.registry.WebsiteSale = publicWidget.Widget.extend(VariantMixin, {
             return;
         }
 
-        var $component = $(ev.currentTarget).closest('.js_product');
-        $component.find('input').each(function () {
-            var $el = $(this);
-            $el.attr('checked', $el.is(':checked'));
-        });
-        $component.find('select option').each(function () {
-            var $el = $(this);
-            $el.attr('selected', $el.is(':selected'));
-        });
+        if (this.hashProcessed) {
```

Comment:
Don't see why ?

Reply:
Else when the page is loaded and the variants change following the hash this part of the code will be triggered for every variant in the hash. We want this part of the code to be triggered only when the user himself make a change in the variants. (after the page is loaded)

Reply:
Indeed, that is why we should not simulate change events :wink: On hash processing, do not trigger `change` but call some methods, and move the relevant code of `onChangeVariant` in those methods.

Reply:
I see what I did wrong... all the component logic should be out and only `this._setUrlHash($component);` should be in the if...